ecr-login/api/client.go

```diff
@@ -98,6 +99,22 @@ func (self *defaultClient) GetCredentials(registry, image string) (*Auth, error)
 	return nil, fmt.Errorf("No AuthorizationToken found for %s", registry)
 }
 
+func (self *defaultClient) ListCredentials() ([]*Auth, error) {
+	log.Debug("Listing Credentials from cache")
+
+	auths := []*Auth{}
+	for _, authEntry := range self.credentialCache.List() {
```

Comment:
We are returning expired credentials here - list just returns the serverUrls and usernames (not credentials).

```
$ /usr/local/bin/docker-credential-ecr-login list
{"<id>.dkr.ecr.us-west-2.amazonaws.com":"AWS"}
```

In the build case, Docker does a list first, then seems to call get for each one. That seems pretty reasonable and much simpler than refreshing ahead of time (since the get will refresh any that are expired). That way if the build process ever gets smarter, it will only refresh what it needs (vs. everything from list).

Totally agree about the filtering for regions - I want to fix that before merging - will require a bit of refactoring in the cache so it's not too much of a hack. The assumed role case isn't one I had thought of, but I wanted to scope down based on access (for the multiple profile case). The cacheKey is "region-hashedAccessKey-registry-id", so we could either move to 1) filtering by the hashedAccessKey only, 2) move from hashedAccessKey to hashedAccountId, and filter by that (if available from SDK), or 3) return all registries without any filtering. Thoughts?